### PR TITLE
Move code from internal repo to public repo

### PR DESCRIPTION
**Pull request by ThishaniLucas**

### CODE REVIEW

File: health.base/Module.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Health Base Module
+
+## Module Overview
+This module is Default module of Health Base package which contains base common capabilities used in other
```
Comment (isuruh15) [inline health.base/Module.md]:
```suggestion
This is the Default module of Health Base package which contains base common capabilities used in other 
```

File: health.base/modules/auth/Module.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Private Key JWT Authentication Provider
+
+# Module Overview
+
+This module contains an authentication provider that supports client assertion JWTs. Using this module, access token requests to token endpoints can be sent with a client assertion. The module handles also the caching and management of access tokens.
```
Comment (isuruh15) [inline health.base/modules/auth/Module.md]:
```suggestion
This module contains an authentication provider that supports client assertion JWTs. Using this module, access token requests to token endpoints can be sent with a client assertion. The module also handles the caching and management of access tokens.
```

File: health.base/modules/auth/pkjwt_auth_provider.bal [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    string cachedAccessToken = tokenCache.getAccessToken();
+    if cachedAccessToken == "" {
+        return retrieveTokenFromEP(config, tokenCache);
+    } else {
+        if !tokenCache.isAccessTokenExpired() {
+            return cachedAccessToken;
+        } else {
+            lock {
+                if !tokenCache.isAccessTokenExpired() {
+                    return tokenCache.getAccessToken();
```
Comment (isuruh15) [inline health.base/modules/auth/pkjwt_auth_provider.bal]: Isn't this return statement unreachable?

Comment (ThishaniLucas) [reply health.base/modules/auth/pkjwt_auth_provider.bal]: No. A race condition may cause the token to be regenerated by the time the lock stmnt is reached. If so, we return the new token from the cache.

**Review by sameeragunarathne — DISMISSED** [review]

File: health.base/modules/auth/pkjwt_auth_provider.bal (hunk 3 above)
Comment (isuruh15) [reply health.base/modules/auth/pkjwt_auth_provider.bal]: understood. thanks

**Review by isuruh15 — DISMISSED** [review]

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**